### ISSUE TITLE
[BUG] - Export Setting not Working

### ISSUE THREAD

**Issue by NJDevils28** [WebTools-NG/WebTools-NG#699]
**Describe the bug**
When exporting my users info to a CSV file, the data is not usable and unintelligible.
When looking at the data in Notepad++ there are no Tabs
When importing with Excel, all the data is in a single cell per row.

First line or heading line looks like this:
**User IDundefinedUser titleundefinedUser nameundefinedUser friendly nameundefinedUser email addressundefinedUser RestrictedundefinedLink to User thumbnailundefinedIs the user part of your HomeundefinedUser invite statusundefine**

All subsequent lines do not contain a {TAB} between fields
All field contain unknown data such as the word 'undefined'
Fields are not encapsulated with Quotes
Here is a sample of a line.  I replaced any PII with "X's":
**NaNXXXXXXXXundefinedXXXXXXXXundefinedN/AundefinedXXXXX.XXXXXXX@hotmail.comundefinedNaNhttps://plex.tv/users/44d4129d9782b69d/avatar?c=1702823901undefinedNaNacceptedundefine**

I am using Version 4.116.1 of PLEX

**To Reproduce**
I navigate to the Export Settings panel.
I fill in the options like so:
Select Date/Time<English>
Date Style<Full>
Time Style<Full>
Delimiter<->
Qualifier<>
What to use<N/A>
Delimiter between columns in CSV<{TAB}>
Replace Return<>
Replace NewLine<>
**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [Windows 11]

**Additional context**

**Logs**
Please provide WebTools-NG logs in a zip
[WebTools-NG.log](https://github.com/WebTools-NG/WebTools-NG/files/13696563/WebTools-NG.log)
[WebTools-NG.log](https://github.com/WebTools-NG/WebTools-NG/files/13696567/WebTools-NG.log)

**Comment by trumpy81** [comment]
Please see this article:

https://github.com/WebTools-NG/WebTools-NG/wiki/Help-%28Hints%2C-Tips-and-Videos%29#to-import-a-csv-file-so-that-the-columns-are-presented-correctly-in-microsoft-excel

Note: you may skip changing **tab** to **|** in the instructions.

**Comment by DrWatson** [comment]
I have seen this too. I have looked at the import instructions above, but that's not the issue. If I look at the raw text file, there are no field delimiters at all. I've tried pipe (|) and tab and comma and none appear in the export.

**Comment by trumpy81** [comment]
@DrWatson 
Please provide a sample output file and your WebTools-NG logs please.

**Comment by DrWatson** [comment]
Here is what the exported file looks like (redacted of course):
`User IDundefinedUser titleundefinedUser nameundefinedUser friendly nameundefinedUser email addressundefinedUser RestrictedundefinedLink to User thumbnailundefinedIs the user part of your HomeundefinedUser invite statusundefine
NaNXXXXXXundefinedXXXundefinedN/AundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedN/AundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedN/AundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedPERSONNAMEundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedN/AundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedN/AundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedN/AundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
NaNXXXXXXundefinedXXXundefinedPERSONNAMEundefinedXXX@XXX.XXXundefinedNaNhttps://plex.tv/users/XXXXXXXXXXXXXXXX/avatar?c=XXXXXXXXXXundefinedNaNacceptedundefine
`

Where do I find the log files...sorry new user.

**Comment by DrWatson** [comment]
nvm. found them. Where can I change the log level? I imagine these logs are not informative enough:
[WebTools-NG.zip](https://github.com/WebTools-NG/WebTools-NG/files/13845086/WebTools-NG.zip)

**Comment by DrWatson** [comment]
nvm. Found them. I bumped to debug level since the regular error logging didn't reveal much. Not sure these are much better, but here you go:
[WebTools-NG.zip](https://github.com/WebTools-NG/WebTools-NG/files/13845134/WebTools-NG.zip)

**Comment by trumpy81** [comment]
Thanks for that, but I would like to see the actual .csv file and logs captured just after the export.

**Comment by DrWatson** [comment]
I have to redact the csv file as I don't want my user's information out in the open. Is it okay if it's an edited csv?

**Comment by trumpy81** [comment]
No, I need to see the raw file. Don't worry, I have no interest in the data the file contains, but I need to observe the raw format of the file.

I will delete the file as soon as I have downloaded it.

**Comment by DrWatson** [comment]
Is there another way to get you the file? I really don't want to post it here.

**Comment by trumpy81** [comment]
Upload the file to Google Drive, OneDrive etc.. and post the link.

Like I said, I will delete the file/link from your post as soon as I download it.

**Comment by DrWatson** [comment]
https://drwatson.net/dl/WebTools-NG%20LogsAndCSV.zip

Let me know when you've grabbed it and I will delete the file on my end.

**Comment by trumpy81** [comment]
The problem is you appear to have no Plex server associated with WebTools-NG. There are no libraries configured or showing in your logs.

You have an invalid separator showing - **ColumnSep: '\t'** in the Export settings.

You need to log into WebTools-NG using the same credentials you use for your Plex server/s (which you appear to have done). You also need to select a server before you enter any of the modules.

If you need any help setting up WebTools-NG then visit the Wiki or Website. See the Global Settings link in the main menu.

Wiki - https://github.com/WebTools-NG/WebTools-NG/wiki
Website - https://webtools-ng.github.io/manual/

**Comment by DrWatson** [comment]
I set the column delimiter to "9" and it replaces it with {TAB}. 
![image](https://github.com/WebTools-NG/WebTools-NG/assets/119038/735ab254-7d5b-4a11-9686-e5fa6bf126e7)
Then I run the export and then you see the result I provided. Not sure where \t is coming from, but I assumed that was a marked up value for "tab". 

I have changed the configuration to be a comma and did an export again. It now puts **tabs** in the file now. 
I configure it to be a pipe, and it still puts **tabs** in. 
So when configuring tabs, it puts nothing in. When putting other things in, it puts tabs in. I can work with this work around, but there's something wrong in this area for sure.

The behaviour observed doesn't change regardless of whether a server is associated with WebTools-NG or not. I have selected the server, then hit reload, then went to the Plex.TV module and the behaviour is the same.

**Comment by trumpy81** [comment]
That is because you input an actual Tab instead of using 9 as I showed you in my previous reply.

You need to remove that Tab (which is hidden) and replace it with the recommended pipe character **|**.

If you are unable to remove the hidden tab character go to Global Settings and click on Factory Reset.

This will close WebTools-NG and when you restart WebTools-NG you will need to reconfigure all settings.

After restoring your settings, try the export again.

**Comment by DrWatson** [comment]
I didn't put an actual tab in. I put a 9 in as the UI says. Just as I leave the page, it updates the value to {TAB}.

I have done a reset factory as suggested leaving all settings as default except setting the export dir:

![image](https://github.com/WebTools-NG/WebTools-NG/assets/119038/2dd0ea77-a94f-4f74-8a64-65a0f864dd19)

The exact same export file is produced with no delimiters.

**Comment by trumpy81** [comment]
May I see the output file and your logs please.

**Comment by DrWatson** [comment]
I have found the set of steps that cause the issue:
start up
factory reset
start up again
log in
set server
go to global settings
then export settings
set export dir
click plex.tv
click export for all
** there are no delimiters in the exported files **

then:
close down
start up again
log in
set server
click plex.tv
click export for all
delimiters are correct with pipes

**Comment by trumpy81** [comment]
OK, I have verified your issue as stated above. I will contact Tommy and ask him to look into the issue.

Be aware though, Tommy is extremely busy with his new Plex duties, so he may not be able to attend to this issue for some time.
